### DISCUSSION TITLE
New script too confusing - need help converting, or downgrading to old firmware

### DISCUSSION THREAD

**Discussion by ykatagiri** [RavenSystem/esp-homekit-devices#2018] [MEPLHAA Scripts]
I tried upgrading one of my less-critical WiFi modules to the new firmware, aware of that there are breaking changes, to see if I can make them work.

The upgrade went OK, but the new "io" declaration is way too confusing - I've read the documentation numerous times, but I'm afraid I don't have the knowledge to properly understand and convert my old script to the new one.
[https://github.com/RavenSystem/esp-homekit-devices/wiki/GPIOs-Configuration](https://github.com/RavenSystem/esp-homekit-devices/wiki/GPIOs-Configuration)

I'm pulling my hair out trying to figure it out, but honestly, I've given up - and just want a simple answer from anyone with the knowledge to convert the following to the new script.  I'm aware there are similar topics regarding stock Sonoff devices, but these are different as they use bare ESP01 modules along with a custom designed handmade circuitry.

1: A relay switch using an ESP01 module, active low at GPIO2
`{"c":{"l":1},"a":[{"t":1,"s":0,"0":{"r":[{"g":2,"v":1}]},"1":{"r":[{"g":2,"v":0}]}}]}`

2: PIR Motion Sensor, Active Low at GPIO3
`{"c":{"l":1},"a":[{"t":12,"i":10,"f1":[{"g":3,"p":1,"i":0,"t":0}]}]}`

Also is there a way to downgrade to the old 11.x firmware, once upgraded to 12.x?
I'm assuming you'll need to remove the ESP01 module and flash it again via serial, but if there are other ways without removing the modules, I'd like to know that as well.

Thanks in advance

**Comment by ykatagiri** [comment]
Reagarding the PIR Motion sensor, I've done some more reading and studied more samples and tried editing the script as shown below, and have made it work.  Still studying how to get the Relay Switch on the new script.

PIR Motion Sensor, Active Low at GPIO3
OLD: `{"c":{"l":1},"a":[{"t":12,"i":10,"f1":[{"g":3,"p":1,"i":0,"t":0}]}]}`
NEW: `{"c":{"io":[[[1],3],[[3],6,1,0]],"l":1},"a":[{"t":12,"i":10,"f1":[{"g":3,"t":0}]}]}`

**Comment by xrust83** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Installation#installing-a-specific-version

https://www.youtube.com/watch?v=sDXl5YFpxr0&t=575s
